pytorch_lightning/metrics/classification/accuracy.py

```
@@ -16,33 +16,52 @@
+    is this is not what you want. In multi-dimensional multi-class case the `mdmc_accuracy` parameters
+    gives you a choice between computing the subset accuracy, or counting each sample on the extra
+    axis separately.
 
-    If preds and target are the same shape and preds is a float tensor, we use the ``self.threshold`` argument.
-    This is the case for binary and multi-label logits.
-
-    If preds has an extra dimension as in the case of multi-class scores we perform an argmax on ``dim=1``.
+    Accepts all input types listed in :ref:`metrics:Input types`.
 
     Args:
         threshold:
-            Threshold value for binary or multi-label logits. default: 0.5
+            Threshold probability value for transforming probability predictions to binary
+            (0,1) predictions, in the case of binary or multi-label inputs. Default: 0.5
+        mdmc_accuracy:
+            Determines how should the extra dimension be handeled in case of multi-dimensional multi-class
+            inputs. Options are ``"global"`` or ``"subset"``.
+
+            If ``"global"``, then the inputs are treated as if the sample (``N``) and the extra dimension
+            were unrolled into a new sample dimension.
+
+            If ``"subset"``, than the equivalent of subset accuracy is performed for each sample on the
+            ``N`` dimension - that is, for the sample to count as correct, all labels on its extra dimension
+            must be predicted correctly (the ``top_k`` option still applies here). The final score is then
+            simply the number of totally correctly predicted samples.
+        top_k:
+            Number of highest probability entries for each sample to convert to 1s, relevant
+            only for (multi-dimensional) multi-class inputs with probability predictions. The
+            default value (``None``) will be interpreted as 1 for these inputs.
```

Why is default not 1?

---

this handles multi-label and binary cases too, so `top_k=1` won't make sense there. Ideally, it should raise an error in such a case. haven't checked yet. if it does.

---

It does raise an error for such cases - through the formatting function from the previous PR :) And yes, I agree with this - making it default to 1 could be potentially misleading in cases where top_k does not really apply.